Classes/WPEditorViewController.m

```diff
@@ -1555,13 +1589,25 @@ - (void)editorView:(WPEditorView*)editorView
 - (void)editorView:(WPEditorView*)editorView
       fieldFocused:(WPEditorField*)field
 {
-    if (!field || field == self.editorView.titleField) {
+    if (field == self.editorView.titleField) {
         [self.toolbarView enableToolbarItems:NO shouldShowSourceButton:YES];
     } else if (field == self.editorView.contentField) {
         [self.toolbarView enableToolbarItems:YES shouldShowSourceButton:YES];
     }
 }
 
+- (void)editorView:(WPEditorView*)editorView sourceFieldFocused:(UIView*)view
+{
+    [self.toolbarView enableToolbarItems:NO shouldShowSourceButton:YES];
+    
+    // Enable the toolbar if the HTML editor has focus
+    if ([view isKindOfClass:[UITextView class]]) {
```

Comment:
The above line took a bit for me to understand properly, and I feel it's prone to breaking if the source view type is changed for any reason.  I feel a more robust approach could be along the lines of:

`view == seld.editorView.sourceView`

Thoughts?